model/labels/matcher.go

```diff
@@ -78,7 +79,30 @@ func MustNewMatcher(mt MatchType, name, val string) *Matcher {
 }
 
 func (m *Matcher) String() string {
-	return fmt.Sprintf("%s%s%q", m.Name, m.Type, m.Value)
+	const quote = 1
+	const matcher = 2
+	// As we're not on go1.22 yet and we don't have the new fancy AvailableBuffer method on strings.Builder,
```

Comment:
Given that v2.52 is already separated from the main branch, and Go is already at 1.22.3, I don't see a problem with requiring Go1.22 from now on. So If it helps with this PR, feel free to bump the requirement.

Reply:
In fact, the RC for v2.52 is already built with Go1.22.

Reply:
Turns out, it's the `bytes.Buffer`, not `strings.Builder` that has the `AvailBuffer` method. I copied the implementation from `Labels.String()` that uses it, with the preallocated array on stack. No need to bump the go version after all.